pkg/provision/container.go

```diff
@@ -217,24 +217,25 @@ func containerDecommission(ctx context.Context, reservation *Reservation) error
 	}
 
 	info, err := container.Inspect(tenantNS, containerID)
-	if err != nil {
-		return errors.Wrapf(err, "failed to inspect container %s", containerID)
-	}
+	if err == nil {
```

Comment:
Why u didn't keep the `if err != nil` and just change the return to a log, and return nil instead of change the code indentation.

Other that that it's a LGTM